NeoML/src/Dnn/Dnn.cpp

```diff
@@ -523,7 +526,10 @@ void CDnn::ForceRebuild()
 
 void CDnn::DeleteLayerImpl( CBaseLayer& layer )
 {
-	NeoAssertMsg( !IsReferenceDnn(), "For ReferenceDnn deleting layers is restricted" );
+	NeoAssertMsg( !IsReferenceDnn()
+		|| dynamic_cast<CCompositeSourceLayer*>( &layer ) != nullptr
```

Comment:
This two lines are often encountered in the code. Could we make some inline function for this check?

By the way, `dynamic_cast` could be a symptom of ill-formed architecture. Do we need it as cast or we could use special virtual method?

This is just a notice. No need to fix it in this Pull Request.